modules/javafx.controls/src/main/java/javafx/scene/chart/AreaChart.java

```diff
@@ -554,6 +555,17 @@ LegendItem createLegendItemForSeries(Series<X, Y> series, int seriesIndex) {
         return legendItem;
     }
 
+    /** {@inheritDoc} */
+    @Override void seriesBeingRemovedIsAdded(Series<X,Y> series) {
+        if (timeline != null) {
+            timeline.setOnFinished(null);
+            timeline.stop();
```

Comment:
> two questions:
> 
>     1. should the timeline object be set to null here?  the reference will get overwritten in seriesRemoved():414, there is probably no need to keep this object in memory
> 
Yes timeline can be made null here. Updated the code.

>     2. is it possible that `seriesBeingRemovedIsAdded()` will be invoked twice with a different `series` argument?  In other words, are we going to face a situation when a `timeline` from unrelated series will not get `stop`'ped?
> 
> 
> (these question apply to other charts as well)

Since `seriesBeingRemovedIsAdded()` is called from the event handler which gets invoked when the series changes and further there is a check on `setToRemove` boolean, I believe the method will not be called with different `series` and cause unwanted `series` timeline to be stopped.

Reply:
Thank you for clarification, @karthikpandelu